cmd/collector/app/server/http.go

```diff
@@ -52,8 +53,8 @@ func StartHTTPServer(params *HTTPServerParams) (*http.Server, error) {
 		IdleTimeout:       params.IdleTimeout,
 		ErrorLog:          errorLog,
 	}
-	if params.TLSConfig.Enabled {
-		tlsCfg, err := params.TLSConfig.Config(params.Logger) // This checks if the certificates are correctly provided
+	if params.TLSConfig != nil {
+		tlsCfg, err := params.TLSConfig.LoadTLSConfig(context.Background())
```

Comment:
below you are creating a listener manually. The point was to use ToListener and ToServer from confighttp

Reply:
the next thing to do after fixing all reviewed issues 👍